packages/teams-js/src/internal/videoUtils.ts

```diff
@@ -78,36 +135,186 @@ function createProcessedStreamGenerator(
+    // (oti1 stands for "one texture input version 1")
+    const ONE_TEXTURE_LAYOUT_ID = 0x6f746931;
+    // Stream id for audio inference metadata, which is the 4-byte ASCII string "1dia" hardcoded by the host
+    // (1dia stands for "audio inference data version 1")
+    const AUDIO_INFERENCE_RESULT_STREAM_ID = 0x31646961;
+
+    // The rectangle of pixels to copy from the texture. The first two rows are the header.
+    const headerRect = { x: 0, y: 0, width: texture.codedWidth, height: 2 };
+    // allocate buffer for the header
+    // it's in NV12 format (https://learn.microsoft.com/en-us/windows/win32/medfound/recommended-8-bit-yuv-formats-for-video-rendering#nv12), so the
+    // number bytes of the header is (the size of the Y plane + the size of the UV plane)
+    // (headerRect.width * headerRect.height) + (headerRect.width * headerRect.height) / 2
+    const headerBuffer = new ArrayBuffer((headerRect.width * headerRect.height * 3) / 2);
+    await texture.copyTo(headerBuffer, { rect: headerRect });
+    const headerDataView = new Uint32Array(headerBuffer);
+    // headerDataView contains the following data:
+    // 0: oneTextureLayoutId
+    // 1: version
+    // 2: frameRowOffset
+    // 3: frameFormat
+    // 4: frameWidth
+    // 5: frameHeight
+    // 6: multiStreamHeaderRowOffset
+    // 7: multiStreamCount
+    if (headerDataView.length < 8) {
+      this.notifyError('Invalid video frame header');
+      throw new Error('Invalid video frame header');
+    }
+    // ensure the texture layout is supported
+    if (headerDataView[0] !== ONE_TEXTURE_LAYOUT_ID) {
+      this.notifyError('Unsupported texture layout');
+      throw new Error('Unsupported texture layout');
+    }
+
+    // The rectangle of pixels to copy from the texture. Metadata are at the bottom.
+    const metadataRect = {
+      x: 0,
+      y: headerDataView[6], // multiStreamHeaderRowOffset
+      width: texture.codedWidth,
+      height: texture.codedHeight - headerDataView[6], // multiStreamHeaderRowOffset
+    };
+    // allocate buffer for the metadata, buffer size = (the size of the Y plane + the size of the UV plane)
+    // (metadataRect.width * metadataRect.height) + (metadataRect.width * metadataRect.height) / 2
+    const metadataBuffer = new ArrayBuffer((metadataRect.width * metadataRect.height * 3) / 2);
+    await texture.copyTo(metadataBuffer, { rect: metadataRect });
+    const metadata = new Uint32Array(metadataBuffer);
+    let audioInferenceResult: Uint8Array | undefined;
+    for (let i = 0, index = 0; i < headerDataView[7] /* multiStreamCount */; i++) {
+      const streamId = metadata[index++];
+      const streamDataOffset = metadata[index++];
+      const streamDataSize = metadata[index++];
+      const streamData = new Uint8Array(metadataBuffer, streamDataOffset, streamDataSize);
+      if (streamId === AUDIO_INFERENCE_RESULT_STREAM_ID) {
```

Comment:
when would streamId be some other value? This seems like the only streamId value you're ever setting

Reply:
This is the only supported metadata type for now.